src/frontend/src/pages/error.jsx

```diff
@@ -0,0 +1,170 @@
+import ArrowBack from '@material-ui/icons/ArrowBack';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import DynamicBackgroundContainer from '../components/DynamicBackgroundContainer';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    fontFamily: 'Roboto',
+    display: 'block',
+    zIndex: 100,
+    padding: theme.spacing(2, 4, 2, 4),
+    position: 'relative',
+    top: '40vh',
+    margin: 'auto',
+    backgroundColor: theme.palette.primary.main,
+    overflow: 'visible',
+    [theme.breakpoints.between('xs', 'sm')]: {
+      top: theme.spacing(24),
+    },
+  },
+  h1: {
+    fontWeight: 'bold',
+    opacity: 0.85,
+    color: theme.palette.grey[100],
+    fontSize: '12rem',
+    [theme.breakpoints.between('xs', 'sm')]: {
+      fontSize: '4rem',
+    },
+    [theme.breakpoints.between('md', 'lg')]: {
+      fontSize: '8rem',
+    },
+    [theme.breakpoints.up('xl')]: {
+      fontSize: '12rem',
+    },
+  },
+  h2: {
+    fontSize: '2rem',
+    color: theme.palette.grey[200],
+    marginTop: '1.75rem',
+    lineHeight: 'inherit',
+    letterSpacing: 'inherit',
+    transition: 'all linear 350ms',
+    [theme.breakpoints.between('xs', 'sm')]: {
+      fontSize: '2rem',
+    },
+    [theme.breakpoints.between('md', 'lg')]: {
+      fontSize: '4rem',
+    },
+    [theme.breakpoints.up('xl')]: {
+      fontSize: '8rem',
+    },
+  },
+  h3: {
+    fontFamily: 'monospace',
+    fontSize: '2rem',
+    padding: theme.spacing(4),
+    color: theme.palette.grey[300],
+    // backgroundColor: theme.palette.error.light,
+    borderRadius: theme.spacing(1),
+  },
+  link: {
+    color: 'white',
+    fontFamily: 'Roboto, sans-serif',
+    textDecoration: 'none',
+    fontSize: '1.5rem',
+    margin: '0 0.5rem 0 0.5rem',
+  },
+  fab: {
+    position: 'relative',
+    fontSize: '1.5rem',
+    color: 'white',
+    bottom: -45,
+    zIndex: 200,
+    backgroundColor: theme.palette.secondary.light,
+    transition: 'all linear 250ms',
+    '&:hover': {
+      backgroundColor: theme.palette.secondary.dark,
+    },
+    [theme.breakpoints.between('xs', 'sm')]: {
+      right: '-2rem',
+    },
+  },
+
+  buttonText: {
+    fontSize: '1.5rem',
+    paddingLeft: theme.spacing(1),
+    [theme.breakpoints.between('xs', 'sm')]: {
+      display: 'none',
+    },
+  },
+}));
+
+function CreateInnerErrorContent(props) {
+  const classes = useStyles();
+
+  const errorMessages = {
+    '400': 'We did not understand the request!',
+    '401': 'You are not authorized to view this page!',
+    '403': 'Access is not allowed for the requested page!',
+    '404': 'We could not find what you were looking for!',
+    '405': 'Method is not allowed!',
+  };
+
+  if (props.status !== '500' && !props.message) {
+    return (
+      <Typography variant="body1" className={classes.h2}>
+        {errorMessages[props.status]}{' '}
```

Comment:
Should you have some kind of fallback "Something went wrong" error message for any case not covered by the error codes above?  We might add more in the future.